packages/app-store/caldavcalendar/api/add.ts

```diff
@@ -40,8 +40,22 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       await prisma.credential.create({
         data,
       });
-    } catch (reason) {
-      logger.error("Could not add this caldav account", reason);
+    } catch (e) {
+      logger.error("Could not add this caldav account", e);
+      if (e instanceof Error) {
+        let message = e.message;
+        if (e.message.indexOf("Invalid credentials") > -1 && url.indexOf("dav.php") > -1) {
```

Comment:
If we know it is baikal and the error is "Invalid credentials", then we introduce the hint

Reply:
nit: `.includes` is easier to eyes :D

Reply:
nit: Might be a good idea to make an abstraction like `isBaikal`. That would not require any comment and in future too it would be clear what's going on